messaging-client-databus/src/main/java/com/inmobi/messaging/consumer/databus/AbstractMessagingDatabusConsumer.java

```diff
@@ -283,7 +288,11 @@ private void checkClosedReaders() throws EndOfStreamException {
   protected synchronized void start() throws IOException {
     createPartitionReaders();
     for (PartitionReader reader : readers.values()) {
+      if (startedReaders.contains(reader)) {
```

Comment:
Consumer has to start all readers and this start() is called only at the time of initialization. So we can safely remove the condition.